### ISSUE TITLE
Lazyload script and styles after .lazyloaded?

### ISSUE THREAD

**Issue by tawg** [aFarkas/lazysizes#431]
Hi.

I see that scripts and styles are lazyloaded at start not after the <div> element with them is in viewport.
Wouldn't  it be a better idea?
Only when we scroll to the element that call these resources they get loaded?

**Comment by aFarkas** [comment]
@tawg 
If you mean the plugin unveilhooks, than here is the answer.

Scripts and styles are only added as soon as the element with the class lazyload is inviewport or near the viewport. The position where scripts/styles are inserted by script isn't important.

Please close if this answers your question or give otherwise more information to your problem.

**Comment by tawg** [comment]
HI @aFarkas 
Than you for your answer.
Yes it's about this plugin.
When I look at you demo page:
https://afarkas.github.io/lazysizes/index.html
And when I didn't scroll down yet to the Carousel area I see in the inspection that "assets/js/bootstrap.min.js" and "assets/css/carousel.css" are already in <head>.
So they are inserted before scrolling into the carouse area. Just after a second or two the DOM get changed.
I tested in latest Chrome in Incognito mode.

**Comment by aFarkas** [comment]
@tawg 
The demo you are reffering to is also demonstrating the use of the lazypreload class.

**Comment by tawg** [comment]
@aFarkas 
Yes I've notices that. I've read about preload function and not the class.
Thanks for clarifying.
